### PR TITLE
Don't use tmp for lock file, use /var/lock

### PR DESCRIPTION
**Pull request by ntascii**
Some clusters have cleanup jobs running to wipe out /tmp periodically. Keeping .deploy-agent.lock file under /tmp would possible have 2 deployd instances running at same time if /tmp is lost. Used /var/lock. 

Tested locally by running two instances at the same time. 
```
Error: deploy-agent may already be running. Only one instance of it can run at a time.
(teletraan) anhn@dev-anhn deploy-agent (no-tmp-for-lock) $ ls -a /var/lock
.  ..  .deploy-agent.lock  asap_checkin.lock  cmdb_checkin.lock  lvm  ntpdate  subsys
```

### CODE REVIEW

File: deploy-agent/tox.ini [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [tox]
-envlist=py27,py37,py38,py39
```
[the remark this answers is not in the document]
Comment (ntascii) [reply deploy-agent/tox.ini:2]: Remove test option for 2.7 as we are using py3 and will deprecate py2 soon.

[the remark this answers is not in the document]
Comment (robbintt) [reply deploy-agent/tox.ini:2]: nit - maybe better to keep the 2 concerns separate, so each PR has an atomic focus

**Review by robbintt — APPROVED** [review]

File: deploy-agent/tox.ini (hunk 1 above)
[the remark this answers is not in the document]
Comment (ntascii) [reply deploy-agent/tox.ini]: yeah, will deprecate py2 code in a separate PR.

**Comment by ntascii** [comment]
Added support for both py2 and py3. Tested on py2 and py3. UT on both py2 and py3.

File: deploy-agent/deployd/common/single_instance.py [hunk 2]
```diff
@@ -17,19 +17,22 @@
...
 import os
 import stat
-import tempfile
+import errno
 import fcntl
 from . import utils
+from future.utils import PY3
 
 log = logging.getLogger(__name__)
+LOCKFILE_DIR = '/var/lock'
```
Comment (robbintt) [inline deploy-agent/deployd/common/single_instance.py:26]: maybe add a note that it must be an absolute path for python 2 or os.makedirs will become confused

**Review by robbintt — DISMISSED** [review]

File: deploy-agent/deployd/common/single_instance.py (hunk 2 above)
Comment (ntascii) [reply deploy-agent/deployd/common/single_instance.py:26]: I actually added a comment regarding `exist_ok` for py2.

**Review by robbintt — APPROVED** [review]

**Pull request merged**